fhir-server/src/main/java/com/ibm/fhir/server/resources/FHIRResource.java

```diff
@@ -600,6 +618,30 @@ public Response conditionalDelete(@PathParam("type") String type) throws Excepti
             log.exiting(this.getClass().getName(), "delete(String,String)");
         }
     }
+    
+    
+    private long parseIfModifiedSince() {
```

Comment:
We can probably use DateTimeHandler from Search's pattern and use a parseBest
Snippet

```
private static final DateTimeFormatter DATE_TIME_PARSER_FORMATTER =
            new DateTimeFormatterBuilder()
                    .appendPattern("yyyy")
                    .optionalStart()
                        .appendLiteral('-')
                        .appendPattern("MM")
                        .optionalStart()
                            .appendLiteral('-')
                            .appendPattern("dd")
                            .optionalStart()
                                .appendLiteral("T")
                                .optionalStart()
                                .appendPattern("HH")
                                .optionalStart()
                                    .appendLiteral(':')
                                    .appendPattern("mm")
                                    .optionalStart()
                                        .appendLiteral(':')
                                        .appendPattern("ss")
                                        .optionalStart()
                                            .appendFraction(ChronoField.MICRO_OF_SECOND, 0, 6, true)
                                        .optionalEnd()
                                    .optionalEnd()
                                .optionalEnd()
                                .optionalStart()
                                    .appendPattern("XXX")
                                .optionalEnd()
                            .optionalEnd()
                            .optionalEnd()
                        .optionalEnd()
                    .optionalEnd()
                    .toFormatter();
    // @formatter:on
```

then one call to 

```
return DATE_TIME_PARSER_FORMATTER.withResolverStyle(ResolverStyle.SMART).parseBest(value,
                    ZonedDateTime::from, LocalDateTime::from, LocalDate::from, YearMonth::from, Year::from);
```

Reply:
en... seems the new pattern is much more complicated then using SimpleDateFormat, but let me try to change to use it ... :)

Reply:
just noticed that it's mainly used for formater with optional elements, but in my case, we need 2 different formaters with the required different elements. so I would prefer to keep current implementation.

Reply:
I don't think you need two formatters.  I think breaking this discussion out and talking with John may be best